reader-osm/src/test/java/com/graphhopper/reader/osm/GraphHopperOSMTest.java

```diff
@@ -149,6 +145,67 @@ public void testLoadOSMNoCH() {
         gh.close();
     }
 
+    @Test
+    public void testQueryLocationIndexWithBBox() {
+        GraphHopper gh = new GraphHopperOSM().setStoreOnFlush(true).
+                setEncodingManager(new EncodingManager("car")).
+                setCHEnabled(false).
+                setGraphHopperLocation("./target/monacotmp-gh").
+                setDataReaderFile("../core/files/monaco.osm.gz");
+        gh.importOrLoad();
+
+        final NodeAccess na = gh.getGraphHopperStorage().getNodeAccess();
+        final Collection<Integer> indexNodeList = new TreeSet<>();
+        LocationIndexTree index = (LocationIndexTree) gh.getLocationIndex();
+        final EdgeExplorer edgeExplorer = gh.getGraphHopperStorage().createEdgeExplorer();
+        final BBox bbox = new BBox(7.422, 7.429, 43.729, 43.734);
+        index.query(bbox, new LocationIndexTree.Visitor() {
+            @Override
+            public void onCellBBox(BBox bbox, int width) {
+            }
+
+            @Override
+            public void onNode(int nodeId) {
+                double lat = na.getLatitude(nodeId);
+                double lon = na.getLongitude(nodeId);
+                if (bbox.contains(lat, lon))
+                    indexNodeList.add(nodeId);
+
+                // TODO should be done behind the scene?
```

Comment:
Will make it possible via a `SimpleVisitor` class or something. Often this iteration has to be done again on the client side e.g. to fetch the edges and so this work would be an unnecessary overhead then.